src/coreml.cs

```diff
@@ -381,5 +425,131 @@ interface MLCustomLayer {
+	[Protocol]
+	interface MLBatchProvider {
+
+		[Abstract]
+		[Export ("count")]
+		nint Count { get; }
+
+		[Abstract]
+		[Export ("featuresAtIndex:")]
+		IMLFeatureProvider GetFeatures (nint index);
+	}
+
+	[Watch (5,0), TV (12,0), Mac (10,14, onlyOn64: true), iOS (12,0)]
+	[Protocol]
+	interface MLCustomModel {
+
+		[Abstract]
+		[Export ("initWithModelDescription:parameterDictionary:error:")]
+		IntPtr Constructor (MLModelDescription modelDescription, NSDictionary<NSString, NSObject> parameters, [NullAllowed] out NSError error);
+
+		[Abstract]
+		[Export ("predictionFromFeatures:options:error:")]
+		[return: NullAllowed]
+		IMLFeatureProvider GetPrediction (IMLFeatureProvider inputFeatures, MLPredictionOptions options, [NullAllowed] out NSError error);
+
+		[Export ("predictionsFromBatch:options:error:")]
+		[return: NullAllowed]
+		IMLBatchProvider GetPredictions (IMLBatchProvider inputBatch, MLPredictionOptions options, [NullAllowed] out NSError error);
+	}
+
+	[Watch (5,0), TV (12,0), Mac (10,14, onlyOn64: true), iOS (12,0)]
+	[BaseType (typeof(NSObject))]
+	interface MLImageSize {
+
+		[Export ("pixelsWide")]
+		nint PixelsWide { get; }
+
+		[Export ("pixelsHigh")]
+		nint PixelsHigh { get; }
+	}
+
+	[Watch (5,0), TV (12,0), Mac (10,14, onlyOn64: true), iOS (12,0)]
+	[BaseType (typeof(NSObject))]
+	interface MLImageSizeConstraint {
+
+		[Export ("type")]
+		MLImageSizeConstraintType Type { get; }
+
+		[Export ("pixelsWideRange")]
+		NSRange PixelsWideRange { get; }
+
+		[Export ("pixelsHighRange")]
+		NSRange PixelsHighRange { get; }
+
+		[Export ("enumeratedImageSizes")]
+		MLImageSize[] EnumeratedImageSizes { get; }
+	}
+
+	[Watch (5,0), TV (12,0), Mac (10,14, onlyOn64: true), iOS (12,0)]
+	[BaseType (typeof(NSObject))]
+	interface MLMultiArrayShapeConstraint {
+
+		[Export ("type")]
+		MLMultiArrayShapeConstraintType Type { get; }
+
+		[Export ("sizeRangeForDimension")]
+		NSValue[] SizeRangeForDimension { get; }
+
+		[Export ("enumeratedShapes")]
+		NSArray<NSNumber>[] EnumeratedShapes { get; }
+	}
+
+	[Watch (5,0), TV (12,0), Mac (10,14, onlyOn64: true), iOS (12,0)]
+	[BaseType (typeof(NSObject))]
+	interface MLSequence {
+
+		[Export ("type")]
+		MLFeatureType Type { get; }
+
+		[Static]
+		[Export ("emptySequenceWithType:")]
+		MLSequence CreateEmpty (MLFeatureType type);
+
+		[Static]
+		[Export ("sequenceWithStringArray:")]
+		MLSequence Create (string[] stringValues);
+
+		[Export ("stringValues")]
+		string[] StringValues { get; }
+
+		[Static]
+		[Export ("sequenceWithInt64Array:")]
+		MLSequence Create (NSNumber[] int64Values);
+
+		[Export ("int64Values")]
+		NSNumber[] Int64Values { get; }
+	}
+
+	[Watch (5,0), TV (12,0), Mac (10,14, onlyOn64: true), iOS (12,0)]
+	[BaseType (typeof(NSObject))]
+	interface MLSequenceConstraint : NSCopying {
+
+		[Export ("valueDescription")]
+		MLFeatureDescription ValueDescription { get; }
+
+		[Export ("countRange")]
+		NSRange CountRange { get; }
```

Comment:
same as before, only exposed thru a read-only property
no point in having a default ctor